File: C++/MEDIUM/3016. Minimum Number of Pushes to Type Word II.cpp
class Solution
{
public:
    int minimumPushes(string word)
    {
        int counting[26] = {0};
        char alphabet[26] = {};
        for (int i = 0; i < 26; i++)
        {
            alphabet[i] = 'a' + i;
        }
        for (char c : word)
        {
            counting[c - 'a']++;
        }
        sort(alphabet, alphabet + 26 * sizeof(char), [counting](char a, char b)
             { return counting[a - 'a'] > counting[b - 'a']; });
        int numCount = 0;
        int plus = 1;
        int answer = 0;
        for (int i = 0; i < 26; i++)
        {
            answer += plus * counting[alphabet[i] - 'a'];
            numCount++;
            if (numCount == 8)
            {
                numCount = 0;
                plus++;
            }
        }
        return answer;
    }
};

File: 3016-minimum-number-of-pushes-to-type-word-ii/3016-minimum-number-of-pushes-to-type-word-ii.java
class Solution {
    public int minimumPushes(String word) {
        int[] cnts = new int[26];
        for (char c : word.toCharArray())
        {
            cnts[c - 'a'] ++;
        }
        int answer = 0;
        Arrays.sort(cnts);
        for (int i = 25; i > 17; i--)
        {
            answer += cnts[i];
        }
        for (int i = 17; i > 9; i--)
        {
            answer += cnts[i] * 2;
        }
        for (int i = 9; i > 1; i--)
        {
            answer += cnts[i] * 3;
        }
        answer += cnts[1] * 4;
        answer += cnts[0] * 4;
        return answer;
    }
}